RX3D does not compile with NVHPC

If I compile NEURON with `-DNRN_ENABLE_RX3D=ON ` and the NVHPC compiler
```
$ nvc++ --version
nvc++ 20.9-0 LLVM 64-bit target on x86-64 Linux -tp skylake
NVIDIA Compilers and Tools
Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
```
then the build fails:
```
$ cmake --build . -- -j 1
[5/29] cd /gpfs/bbp.cscs.ch/project/proj16/olupton/nrn/build_olliexternalcnrn/share/lib/python/neuro...hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/python-3.8.3-suxrst/bin/python3
FAILED: share/lib/python/neuron/rxd/geometry3d/CMakeFiles/rx3dextensions_0
cd /gpfs/bbp.cscs.ch/project/proj16/olupton/nrn/build_olliexternalcnrn/share/lib/python/neuron/rxd/geometry3d && bash /gpfs/bbp.cscs.ch/project/proj16/olupton/nrn/build_olliexternalcnrn/share/lib/python/neuron/rxd/geometry3d/runpy.sh /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/python-3.8.3-suxrst/bin/python3
+ echo runpy /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/python-3.8.3-suxrst/bin/python3
runpy /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/python-3.8.3-suxrst/bin/python3
+ pyexe=/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/python-3.8.3-suxrst/bin/python3
+ mingw=
+ shift
+ test x = x1
+ /gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/python-3.8.3-suxrst/bin/python3 setup.py build_ext --build-lib=/gpfs/bbp.cscs.ch/project/proj16/olupton/nrn/build_olliexternalcnrn/lib/python
/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/nvhpc-20.9-rqj5oq/Linux_x86_64/20.9/compilers/bin/nvc++
running build_ext
building 'neuron.rxd.geometry3d.graphicsPrimitives' extension
/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/nvhpc-20.9-rqj5oq/Linux_x86_64/20.9/compilers/bin/nvc++ -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -fPIC -I/gpfs/bbp.cscs.ch/project/proj16/olupton/nrn/share/lib/python/neuron/rxd/geometry3d -I. -I/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/libraries/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/py-numpy-1.19.4-upzqna/lib/python3.8/site-packages/numpy/core/include -I/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/python-3.8.3-suxrst/include/python3.8 -c graphicsPrimitives.cpp -o build/temp.linux-x86_64-3.8/graphicsPrimitives.o -O0
nvc++-Error-Unknown switch: -Wno-unused-result
nvc++-Error-Unknown switch: -Wsign-compare
nvc++-Error-Unknown switch: -fwrapv
error: command '/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0/nvhpc-20.9-rqj5oq/Linux_x86_64/20.9/compilers/bin/nvc++' failed with exit status 1
ninja: build stopped: subcommand failed.
```
cc: @pramodk

--- a/share/lib/python/neuron/rxd/geometry3d/setup.py.in
+++ b/share/lib/python/neuron/rxd/geometry3d/setup.py.in
@@ -3,6 +3,8 @@
 
 import sys
 nrn_srcdir = "@NRN_SRCDIR@"
+@USING_PGI_COMPILER_TRUE@using_pgi=True
+@USING_PGI_COMPILER_FALSE@using_pgi=False
 @BUILD_MINGW_TRUE@mingw = 1
 @BUILD_MINGW_FALSE@mingw = 0
 instdir = "@prefix@"
@@ -11,6 +13,8 @@ if nrn_srcdir[0] == '/' or (len(nrn_srcdir) > 2 and nrn_srcdir[1] == ':'):
 else: # not an absolute path
     # TODO: is this right?
     nrn_srcdir = '../../../../../../' + nrn_srcdir
+
+pgi_compiler_flags = "-noswitcherror"
 
 from distutils.core import setup
 from distutils.extension import Extension
@@ -55,6 +59,7 @@ else:
         define_macros=[('MS_WIN64', None)]
 
     extra_compile_args = ['-O'+olevel] if olevel != "" else []
+    if using_pgi: extra_compile_args.append(pgi_compiler_flags)
 
     include_dirs = [nrn_srcdir + '/share/lib/python/neuron/rxd/geometry3d', '.', numpy.get_include()]
 

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -122,9 +122,10 @@ if(NRN_ENABLE_PYTHON AND PYTEST_FOUND)
       NAME rxdmod_tests
       MODFILE_PATTERNS test/rxd/ecs/*.mod
       SCRIPT_PATTERNS test/rxd/*.py test/rxd/testdata/*.dat)
-    if(NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
-      # This test includes a comparison with a binary blob; Intel builds typically have too-large
-      # numerical differences with the stored values.
+    # This test includes a comparison with a binary blob; Intel and PGI/NVHPC
+    # builds typically have too-large numerical differences with the stored values.
+    set(compiler_blacklist "Intel" "PGI" "NVHPC")
+    if(NOT ${CMAKE_CXX_COMPILER_ID} IN_LIST compiler_blacklist)
       nrn_add_test(
         GROUP rxdmod_tests
         NAME rxd_tests

